CMakeLists.txt

```diff
@@ -1,19 +1,309 @@
+    get_property(MSVC_RUNTIME_LIBRARY_DEFAULT TARGET ${PROJECT_NAME} PROPERTY MSVC_RUNTIME_LIBRARY)
+    string(CONCAT "MSVC_RUNTIME_LIBRARY_STR"
+        $<$<CONFIG:Debug>:
+            MultiThreadedDebug
+        >
+        $<$<CONFIG:Release>:
+            MultiThreaded
+        >
+        $<$<NOT:$<OR:$<CONFIG:Debug>,$<CONFIG:Release>>>:${MSVC_RUNTIME_LIBRARY_DEFAULT}>
+    )
+    set_target_properties(${PROJECT_NAME} PROPERTIES MSVC_RUNTIME_LIBRARY ${MSVC_RUNTIME_LIBRARY_STR})
+endif()
+
+
+# Include directories
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+    "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+
+
+# Compile definitions
+
+target_compile_definitions(${PROJECT_NAME} PRIVATE
+    "$<$<CONFIG:Debug>:"
+        "_DEBUG;"
+        "DEBUG"
+    ">"
+    "$<$<CONFIG:Release>:"
+        "NDEBUG"
+    ">"
+)
+
+if(MSVC)
+    target_compile_definitions(${PROJECT_NAME} PRIVATE
+        "WIN32;"
+        "_WIN64;"
+        "WIN64;"
+        "_WINDOWS;"
+        "UNICODE;"
+        "_UNICODE"
+    )
+endif()
+
+
+# Compile and link options
+
+if(MSVC)
+    target_compile_options(${PROJECT_NAME} PRIVATE
+        $<$<CONFIG:Debug>:
+            /Od;             # Disable optimization
+            /RTC1            # Enables stack frame run-time error checking and reporting when a variable is used without having been initialized.
+        >
+        $<$<CONFIG:Release>:
+            /MP;             # Build with multiple processes
+            /O2;             # Optimize for speed
+            /GF              # Enable string pooling
+        >
+        /Gy;                 # Link per-function
+        /W3;                 # Warning level
+        /Zi;                 # Emit debug info in a separate PDB
+        /TC;                 # Compile all source files as C source code regardles of extension
+        /wd4996;             # Suppress deprecation warnings
+        ${DEFAULT_CXX_EXCEPTION_HANDLING};
+        /GS;                 # Enables security checks against buffer overruns
+        /Y-                  # Disable precompiled headers
+    )
+    target_link_options(${PROJECT_NAME} PRIVATE
+        $<$<CONFIG:Debug>:
+            /INCREMENTAL     # Enables incremental linking (faster builds, larger filesize)
+        >
+        $<$<CONFIG:Release>:
+            /OPT:REF;        # Don't link unused functions
+            /OPT:ICF;        # Remove duplicate function definitions
+            /INCREMENTAL:NO  # Disables incremental linking
+        >
+        /MANIFEST;           # Generates a manifest file
+        /DEBUG:FULL;         # Generates debugging symbols (in a separate PDB file)
+        /MACHINE:${CMAKE_VS_PLATFORM_NAME};
+        /SUBSYSTEM:CONSOLE;  # Not a driver or GUI program
+        /NXCOMPAT;           # Support Windows Data Execution Prevention
+        /DYNAMICBASE         # Use address space layout randomization
+    )
+
+    # Link with setargv
+
+    target_link_options(${PROJECT_NAME} PRIVATE
+        setargv.obj
```

Comment:
This is super important otherwise wildcards don't work on Windows (`jpegoptim *.jpg`)